prediction_market_agent/agents/microchain_agent/functions.py

```diff
@@ -143,31 +138,27 @@ def example_args(self) -> list[t.Union[str, float]]:
         return [get_example_market_id(self.market_type), 2.3]
 
     def __call__(self, market_id: str, amount: float) -> str:
-        outcome_bool = get_boolean_outcome(self.outcome)
+        market: AgentMarket = self.market_type.market_class.get_binary_market(market_id)
+        account_balance = float(get_balance(market_type=self.market_type).amount)
```

Comment:
Question: Do we want to get rid of Decimals and just use floats everywhere? It seemed like a good idea in the beginning, but now it feels it brings overhead without clear benefits. I volunteer to refactor in another PR.

Reply:
Yeah I think that's a good idea. Seems like float precision is sufficient for our use cases - though we weren't to know it when starting out

Reply:
@coderabbit Please create an issue for refactoring Decimals to Floats in prediction-market-agent-tooling repository

Reply:
@coderabbitai  You have permisions for prediction-market-agent-tooling repository, so please will you create that issue about removing Decimal in favor of Floats?

Reply:
Wohoo! Good rabbit!

Reply:
@coderabbit wait whaaat? You said you couldn't but then @kongzii just intimidated you into making you do it. It's important to stick by your principles - if he told you to delete this whole repo would you do it?